### PR TITLE
Added NFS Mount Section

### PR DESCRIPTION
**Pull request by JaimeMagiera**
Recloned from main and reapplied changes to storage.adoc to clean things up.

### CODE REVIEW

File: modules/ROOT/pages/storage.adoc [hunk 1]
```diff
@@ -473,3 +473,81 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
 === Version selection and bootup
 
 A GRUB menu entry is created for each version of Fedora CoreOS currently available on a system. This menu entry references an `ostree` deployment which consist of a Linux kernel, an initramfs and a hash linking to an `ostree` commit (passed via the `ostree=` kernel argument). During bootup, `ostree` will read this kernel argument to determine which deployment to use as the root filesystem. Each update or change to the system (package installation, addition of kernel arguments) creates a new deployment. This enables rolling back to a previous deployment if the update causes problems.
+
+== Network Storage
+
+Fedora CoreOS systems can be configured to mount network storage such as NFS and CIFS. This is best achieved by using Ignition to create systemd units. Shares can be mounted on boot by creating a standard mount unit. Alternately, a share can be mounted when users access the mountpoint by creating an additional automount unit. Below are examples of each for an NFS share.
```
[the remark this answers is not in the document]
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: swapped network filesystems for shares.

File: modules/ROOT/pages/storage.adoc [hunk 2]
```diff
@@ -473,3 +473,81 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
 === Version selection and bootup
 
 A GRUB menu entry is created for each version of Fedora CoreOS currently available on a system. This menu entry references an `ostree` deployment which consist of a Linux kernel, an initramfs and a hash linking to an `ostree` commit (passed via the `ostree=` kernel argument). During bootup, `ostree` will read this kernel argument to determine which deployment to use as the root filesystem. Each update or change to the system (package installation, addition of kernel arguments) creates a new deployment. This enables rolling back to a previous deployment if the update causes problems.
+
+== Network Storage
```
[the remark this answers is not in the document]
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Title modified and section moved.

File: modules/ROOT/pages/storage.adoc [hunk 3]
```diff
@@ -473,3 +473,81 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
...
+
+Fedora CoreOS systems can be configured to mount network storage such as NFS and CIFS. This is best achieved by using Ignition to create systemd units. Shares can be mounted on boot by creating a standard mount unit. Alternately, a share can be mounted when users access the mountpoint by creating an additional automount unit. Below are examples of each for an NFS share.
+
+=== NFS Mounts
+
+.Creating a systemd unit to mount an NFS share on boot.
```
[the remark this answers is not in the document]
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: switched and trailing period removed.

File: modules/ROOT/pages/storage.adoc [hunk 4]
```diff
@@ -473,3 +473,81 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
...
+
+.Creating a systemd unit to mount an NFS share on boot. 
+[source,yaml]
+----
+variant: fcos
+version: 1.3.0
+
+systemd: 
+  units:
+    - name: var-mnt-data.mount
```
[the remark this answers is not in the document]
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: comment added.

File: modules/ROOT/pages/storage.adoc [hunk 5]
```diff
@@ -473,3 +473,81 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
 === Version selection and bootup
 
 A GRUB menu entry is created for each version of Fedora CoreOS currently available on a system. This menu entry references an `ostree` deployment which consist of a Linux kernel, an initramfs and a hash linking to an `ostree` commit (passed via the `ostree=` kernel argument). During bootup, `ostree` will read this kernel argument to determine which deployment to use as the root filesystem. Each update or change to the system (package installation, addition of kernel arguments) creates a new deployment. This enables rolling back to a previous deployment if the update causes problems.
+
+== Network Storage
+
+Fedora CoreOS systems can be configured to mount network storage such as NFS and CIFS. This is best achieved by using Ignition to create systemd units. Shares can be mounted on boot by creating a standard mount unit. Alternately, a share can be mounted when users access the mountpoint by creating an additional automount unit. Below are examples of each for an NFS share.
+
+=== NFS Mounts
```
[the remark this answers is not in the document]
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Title expanded.

File: modules/ROOT/pages/storage.adoc [hunk 6]
```diff
@@ -473,3 +473,81 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
...
+        Requires=network-online.target
+        After=network-online.target
+
+        [Mount]
+        What=example.server.edu:/data
+        Where=/var/mnt/data
+        Type=nfs4
+        Options=_netdev
+
+        [Install]
```
[the remark this answers is not in the document]
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Removed _netdev. Not needed under systemd.

File: modules/ROOT/pages/storage.adoc [hunk 7]
```diff
@@ -354,6 +354,70 @@ storage:
       with_mount_unit: true
 ----
 
+== Adding Network Storage
```
Comment (bgilbert) [inline modules/ROOT/pages/storage.adoc:357]: Nit: we generally don't use title case.

File: modules/ROOT/pages/storage.adoc [hunk 8]
```diff
@@ -354,6 +354,70 @@ storage:
...
+
+Fedora CoreOS systems can be configured to mount network file systems such as NFS and CIFS. This is best achieved by using Ignition to create systemd units. File systems can be mounted on boot by creating a standard mount unit. Alternately, a file system can be mounted when users access the mountpoint by creating an additional automount unit. Below are examples of each for an NFS file system.
```
Comment (bgilbert) [inline modules/ROOT/pages/storage.adoc:359]: nit: we generally treat `filesystem` as one word.

File: modules/ROOT/pages/storage.adoc [hunk 9]
```diff
@@ -354,6 +354,70 @@ storage:
...
+
+=== Configuring NFS Mounts
+
+Creating a systemd unit to mount an NFS file system on boot. Note: the .mount file must be named based on the path (e.g. /var/mnt/data = var-mnt-data.mount)
```
Comment (bgilbert) [inline modules/ROOT/pages/storage.adoc:363]: The leading periods on these titles got dropped, so they're no longer being treated as titles of the examples.

The Note could be moved to a callout using [admonition](https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#admonitions) syntax.

`.mount`, `/var/mnt/data`, and `var-mnt-data.mount` should probably all be monospaced.

File: modules/ROOT/pages/storage.adoc [hunk 10]
```diff
@@ -354,6 +354,70 @@ storage:
...
+systemd: 
+  units:
+    - name: var-mnt-data.mount
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Mount data directory
+
+        [Mount]
+        What=example.server.edu:/data
```
Comment (bgilbert) [inline modules/ROOT/pages/storage.adoc:378]: `example.com` (or org, or net) is [reserved](https://www.rfc-editor.org/rfc/rfc2606.html) for this sort of use.

File: modules/ROOT/pages/storage.adoc [hunk 11]
```diff
@@ -354,6 +354,70 @@ storage:
...
+        Where=/var/mnt/data
+        Type=nfs4
+
+        [Install]
+        WantedBy=remote-fs.target
```
Comment (bgilbert) [inline modules/ROOT/pages/storage.adoc:404]: In the automount case, shouldn't this not be enabled by default?

File: modules/ROOT/pages/storage.adoc [hunk 12]
```diff
@@ -354,6 +354,70 @@ storage:
...
+      contents: |
+        [Unit]
+        Description=Automount data directory
+
+        [Automount]
+        TimeoutIdleSec=20min
+        Where=/var/mnt/data
+
+        [Install]
+        WantedBy=multi-user.target
```
Comment (bgilbert) [inline modules/ROOT/pages/storage.adoc:417]: I _think_ this also wants to be `remote-fs.target`, but I'm not completely sure.  There's a risk of services in `multi-user.target` trying to use the mountpoint before the automounter is up.

**Review by bgilbert — COMMENTED** [review]
Content generally LGTM!  Some typographic nits, and a couple questions about unit installation.

File: modules/ROOT/pages/storage.adoc (hunk 7 above)
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Changed from title case.

File: modules/ROOT/pages/storage.adoc (hunk 8 above)
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Changed to one word in all locations.

File: modules/ROOT/pages/storage.adoc (hunk 9 above)
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Added Admonition syntax.

File: modules/ROOT/pages/storage.adoc (hunk 10 above)
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc]: Switched to example.org

File: modules/ROOT/pages/storage.adoc (hunk 12 above)
Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc:417]: Is correct.

**Review by travier — APPROVED** [review]
LGTM

File: modules/ROOT/pages/storage.adoc [hunk 13]
```diff
@@ -471,3 +535,4 @@ The only supported writable locations are `/etc` and `/var`. `/etc` should conta
 === Version selection and bootup
 
 A GRUB menu entry is created for each version of Fedora CoreOS currently available on a system. This menu entry references an `ostree` deployment which consist of a Linux kernel, an initramfs and a hash linking to an `ostree` commit (passed via the `ostree=` kernel argument). During bootup, `ostree` will read this kernel argument to determine which deployment to use as the root filesystem. Each update or change to the system (package installation, addition of kernel arguments) creates a new deployment. This enables rolling back to a previous deployment if the update causes problems.
+
```
Comment (travier) [inline modules/ROOT/pages/storage.adoc:538]: Minor: Seem unrelated?

Comment (JaimeMagiera) [reply modules/ROOT/pages/storage.adoc:538]: Seems so to me, but it was there before so I figured there must be some logic/history. Ask about retaining it or not at the meeting?

Comment (travier) [reply modules/ROOT/pages/storage.adoc:538]: Sorry, I meant the additional line added in this commit.

Comment (travier) [inline modules/ROOT/pages/storage.adoc:538]:
```suggestion
```

File: modules/ROOT/pages/storage.adoc [hunk 14]
```diff
@@ -354,6 +354,70 @@ storage:
...
+        [Automount]
+        TimeoutIdleSec=20min
+        Where=/var/mnt/data
+
+        [Install]
+        WantedBy=multi-user.target
+----
+
+
```
Comment (travier) [inline modules/ROOT/pages/storage.adoc:420]:
```suggestion
```

**Review by travier — APPROVED** [review]

**Comment by travier** [comment]
Squashed, rebased and pushed as https://github.com/coreos/fedora-coreos-docs/pull/324 as I could not push to this PR as you are using the main branch of your fork.

**Comment by travier** [comment]
Thanks for the doc update.
